patches/server/1045-Vault-API.patch

```diff
@@ -0,0 +1,278 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Boy <sivertpaulsen2@gmail.com>
+Date: Wed, 24 Jul 2024 13:53:58 +0200
+Subject: [PATCH] Vault API
+
+
+diff --git a/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java b/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java
+index 30544783d959fbdf4b6fd53c10bd3806e718a066..5b245cca7c9e7a141b2c1b7f145930b67bf75171 100644
+--- a/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java
++++ b/src/main/java/net/minecraft/world/level/block/entity/vault/VaultServerData.java
+@@ -19,12 +19,12 @@ public class VaultServerData {
+     static final String TAG_NAME = "server_data";
+     static Codec<VaultServerData> CODEC = RecordCodecBuilder.create(
+         instance -> instance.group(
+-                    UUIDUtil.CODEC_LINKED_SET.lenientOptionalFieldOf("rewarded_players", Set.of()).forGetter(data -> data.rewardedPlayers),
+-                    Codec.LONG.lenientOptionalFieldOf("state_updating_resumes_at", Long.valueOf(0L)).forGetter(data -> data.stateUpdatingResumesAt),
+-                    ItemStack.CODEC.listOf().lenientOptionalFieldOf("items_to_eject", List.of()).forGetter(data -> data.itemsToEject),
+-                    Codec.INT.lenientOptionalFieldOf("total_ejections_needed", Integer.valueOf(0)).forGetter(data -> data.totalEjectionsNeeded)
+-                )
+-                .apply(instance, VaultServerData::new)
++                UUIDUtil.CODEC_LINKED_SET.lenientOptionalFieldOf("rewarded_players", Set.of()).forGetter(data -> data.rewardedPlayers),
++                Codec.LONG.lenientOptionalFieldOf("state_updating_resumes_at", Long.valueOf(0L)).forGetter(data -> data.stateUpdatingResumesAt),
++                ItemStack.CODEC.listOf().lenientOptionalFieldOf("items_to_eject", List.of()).forGetter(data -> data.itemsToEject),
++                Codec.INT.lenientOptionalFieldOf("total_ejections_needed", Integer.valueOf(0)).forGetter(data -> data.totalEjectionsNeeded)
++            )
++            .apply(instance, VaultServerData::new)
+     );
+     private static final int MAX_REWARD_PLAYERS = 128;
+     private final Set<UUID> rewardedPlayers = new ObjectLinkedOpenHashSet<>();
+@@ -52,13 +52,15 @@ public class VaultServerData {
+         return this.lastInsertFailTimestamp;
+     }
+ 
+-    Set<UUID> getRewardedPlayers() {
++    // Paper start - Vault API
++    public Set<UUID> getRewardedPlayers() {
```

Comment:
You should use Access Transformers for this and also other occurences below. It will reduce the diff quite a bit...: https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md#access-transformers

Reply:
Think this is done now 👍